src/Controls/tests/DeviceTests/HandlerTestBase.cs

```diff
@@ -220,7 +222,16 @@ protected Task CreateHandlerAndAddToWindow<THandler>(IElement view, Func<THandle
 						IView content = window.Content;
 
 						if (content is IPageContainer<Page> pc)
+						{
 							content = pc.CurrentPage;
+							if (content == null)
+							{
+								await Task.Delay(100);
+								content = pc.CurrentPage;
+							}
+
+							_ = content ?? throw new InvalidOperationException("Current Page Not Initialized");
+						}
```

Comment:
This is interesting... You seem to be getting the content, but if it is some multi-page page, you get the current page... and if that does not work you try again?

Reply:
This is mainly a timing issue with Shell. Once the `OnNavigated` events on shell are timed better then I can get rid of this.

Basically the `CurrentPage` on shell isn't initialized until it's actually navigated to on shell because it's a datatemplate. The CurrentPage doesn't come into existence until the platform requests it.

The initial `Navigated` events on shell all fire a bit too early as well. Ideally I'd just use that instead of having to add a delay.

I'll add a comment